### PR TITLE
Implement file component

### PR DESCRIPTION
**Pull request by sergei-maertens**
Partly fixes open-formulieren/open-forms#2958

Implement the file component in the new form builder, which scopes the translations to the component itself and should resolve these translation issues.

- [x] Run makemessages and ship translations

### CODE REVIEW

File: src/registry/file/edit-validation.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  z.object({
+    of: ofSchema.optional(),
+    maxNumberOfFiles: z.union([z.null(), z.number().int().positive().optional()]),
+    fileMaxSize: buildFileMaxSizeSchema(intl).optional(),
+  });
+
+/**
+ * @todo implement validations:
+ *
+ * - validate fileMaxSize <= serverUploadLimit from context, if set. -> pass builder
```
Comment (sergei-maertens) [inline src/registry/file/edit-validation.ts:40]: TODO

File: src/registry/file/edit.tsx [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+ *
+ * @todo options.withCredentials: true seems to be set somewhere -> session cookie
+ *   needs to be sent by the SDK/client!
+ */
+EditForm.defaultValues = {
+  storage: 'url',
+  url: '',
+  // TODO -> add to type definition!
+  // options: '{"withCredentials": true}',
+  // webcam: false,
```
Comment (sergei-maertens) [inline src/registry/file/edit.tsx:146]: TODO

File: src/registry/file/edit.tsx [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+  // Advanced tab
+  conditional: {
+    show: undefined,
+    when: '',
+    eq: '',
+  },
+  // Validation tab
+  validate: {
+    required: false,
+  },
```
Comment (sergei-maertens) [inline src/registry/file/edit.tsx:169]: TODO: remove `plugins` from type definition (it is not supported for file component)

File: src/registry/file/file-tab.tsx [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+import {useFormikContext} from 'formik';
+import isEqual from 'lodash.isequal';
+import {useContext} from 'react';
+import {FormattedMessage, useIntl} from 'react-intl';
+import useAsync from 'react-use/esm/useAsync';
+
+import {Checkbox, Component, NumberField, Select, TextField} from '@/components/formio';
+import {BuilderContext} from '@/context';
+
+// // lifted from formio's file component in 4.13.x
```
Comment (sergei-maertens) [inline src/registry/file/file-tab.tsx:11]: TODO - use this or something similar to validate the file upload size limit against the server configuration value

File: src/registry/file/file-tab.tsx [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+  const {values} = useFormikContext<FileComponentSchema>();
+  const applyResize = values?.of?.image?.resize?.apply ?? false;
+
+  return (
+    <>
+      <ImageResizeApply />
+      {applyResize && (
+        <Component type="columns">
+          <div className="columns">
+            <div className="column column--span-md">
```
Comment (sergei-maertens) [inline src/registry/file/file-tab.tsx:215]: column styles are in the backend CSS, so they're displayed in a single column in storybook

**Review by SilviaAmAm — APPROVED** [review]

**Pull request merged**